source/common/grpc/async_client_manager_impl.cc

```diff
@@ -24,17 +24,23 @@ AsyncClientFactoryImpl::AsyncClientFactoryImpl(Upstream::ClusterManager& cm,
 
 AsyncClientManagerImpl::AsyncClientManagerImpl(Upstream::ClusterManager& cm,
                                                ThreadLocal::Instance& tls)
-    : cm_(cm), tls_(tls) {}
+    : cm_(cm), tls_(tls) {
+#ifdef ENVOY_GOOGLE_GRPC
```

Comment:
I think we should probably split into two different concrete implementations at this point. Not super crazy abotu the ifdefs and things like `google_tls_slot`, etc. I think as the logic gets more and more complicated it will be easier to reason about to just split them now?

Reply:
The problem there is that we can dynamically switch between using the two different gRPC client libraries in the API. Right now, the `AsyncClientManager` hides this, you just feed it config. So, if we want to split implementation, we'd need to export out from `ClusterManager` a getter for both of them and we'll still need a function to figure out from the config which one to use. I can do this, but it feels a bit cleaner to encapsulate all the multi-client logic in the single module here.

Reply:
Ah OK. Alright that's fine.